From 924921bbc2d9a5caff6d73d887fb56a566246a20 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Fri, 21 May 2021 16:56:28 +0800
Subject: [PATCH] ruy: fix numeric_limits is not a member of std

|external/ruy/ruy/block_map.cc:375:25: error: 'numeric_limits' is not a
member of 'std'
  375 |   int best_score = std::numeric_limits<int>::min();

Upstream-Status: Pending

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 ...umeric_limits-is-not-a-member-of-std.patch | 31 +++++++++++++++++++
 third_party/ruy/workspace.bzl                 |  6 ++++
 2 files changed, 37 insertions(+)
 create mode 100644 third_party/ruy/0001-fix-numeric_limits-is-not-a-member-of-std.patch

diff --git a/third_party/ruy/0001-fix-numeric_limits-is-not-a-member-of-std.patch b/third_party/ruy/0001-fix-numeric_limits-is-not-a-member-of-std.patch
new file mode 100644
index 00000000000..180f6c4cf35
--- /dev/null
+++ b/third_party/ruy/0001-fix-numeric_limits-is-not-a-member-of-std.patch
@@ -0,0 +1,31 @@
+From 3381c9aabda08f08a3871fb67048b7f2d97480ef Mon Sep 17 00:00:00 2001
+From: Hongxu Jia <hongxu.jia@windriver.com>
+Date: Fri, 21 May 2021 16:48:17 +0800
+Subject: [PATCH] fix numeric_limits is not a member of std
+ 
+|external/ruy/ruy/block_map.cc:375:25: error: 'numeric_limits' is not a
+member of 'std'
+  375 |   int best_score = std::numeric_limits<int>::min();
+
+Upstream-Status: Pending
+
+Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
+---
+ ruy/block_map.cc | 1 +
+ 1 file changed, 1 insertion(+)
+
+diff --git a/ruy/block_map.cc b/ruy/block_map.cc
+index 44e5039..d732d1b 100755
+--- a/ruy/block_map.cc
++++ b/ruy/block_map.cc
+@@ -16,6 +16,7 @@ limitations under the License.
+ #include "ruy/block_map.h"
+ 
+ #include <algorithm>
++#include <limits>
+ #include <cstdint>
+ 
+ #ifdef RUY_MAKEBLOCKMAP_DEBUG
+-- 
+2.27.0
+
diff --git a/third_party/ruy/workspace.bzl b/third_party/ruy/workspace.bzl
index c2b8f0531de..91d1243a0c7 100644
--- a/third_party/ruy/workspace.bzl
+++ b/third_party/ruy/workspace.bzl
@@ -2,11 +2,17 @@
 
 load("//third_party:repo.bzl", "third_party_http_archive")
 
+# Sanitize a dependency so that it works correctly from code that includes
+# TensorFlow as a submodule.
+def clean_dep(dep):
+    return str(Label(dep))
+
 def repo():
     third_party_http_archive(
         name = "ruy",
         sha256 = "d8f9dc52c0a52c8470e2e0b60bc16cba91853d812846c075f7ed8404990b003d",
         strip_prefix = "ruy-5bb02fbf90824c2eb6cd7418f766c593106a332b",
+        patch_file = clean_dep("//third_party/ruy:0001-fix-numeric_limits-is-not-a-member-of-std.patch"),
         urls = [
             "https://storage.googleapis.com/mirror.tensorflow.org/github.com/google/ruy/archive/5bb02fbf90824c2eb6cd7418f766c593106a332b.zip",
             "https://github.com/google/ruy/archive/5bb02fbf90824c2eb6cd7418f766c593106a332b.zip",
-- 
2.27.0

